src/vivarium/framework/results/manager.py

```diff
@@ -93,36 +86,38 @@ def on_post_setup(self, _: Event) -> None:
             stratification.name for stratification in registered_stratifications
         )
 
+        # Initialize missing and unused stratification dictionaries for batch-logging
         missing_stratifications = {}
         unused_stratifications = registered_stratification_names.copy()
 
+        kwargs = {
+            "registered_stratifications": registered_stratifications,
+            "registered_stratification_names": registered_stratification_names,
+            "missing_stratifications": missing_stratifications,
+            "unused_stratifications": unused_stratifications,
+        }
+
         for event_name in self._results_context.observations:
             for (
                 _pop_filter,
-                all_requested_stratification_names,
+                requested_stratification_names,
             ), observations in self._results_context.observations[event_name].items():
                 for observation in observations:
                     measure = observation.name
-                    if all_requested_stratification_names is not None:
-                        df, unused_stratifications = self._initialize_stratified_results(
-                            measure,
-                            all_requested_stratification_names,
-                            registered_stratifications,
-                            registered_stratification_names,
-                            missing_stratifications,
-                            unused_stratifications,
-                        )
-                    else:
-                        # Initialize a completely empty dataframe
-                        df = pd.DataFrame()
+                    kwargs["requested_stratification_names"] = requested_stratification_names
+
+                    df, unused_stratifications = observation.results_initializer(
+                        measure, **kwargs
```

Comment:
There's no reason to create a `kwargs` dictionary here. It just makes the code more confusing to read.